hepdata_validator/data_file_validator.py

```diff
@@ -163,58 +174,72 @@ def check_independent_variable_values(self, file_path, data_item):
-                                    error_plus = error_minus = error['symerror']
-                                elif 'asymerror' in error:
-                                    error_plus = error['asymerror']['plus']
-                                    error_minus = error['asymerror']['minus']
-
-                                error_plus = convert_to_float(error_plus)
-                                error_minus = convert_to_float(error_minus)
-
-                                if error_plus == 0 and error_minus == 0:
-                                    zero_uncertainties.append(True)
-                                else:
-                                    zero_uncertainties.append(False)
-
-                            if len(zero_uncertainties) > 0 and all(zero_uncertainties):
+        for dependent_variable in data['dependent_variables']:
+            for i, value in enumerate(dependent_variable['values']):
+                if 'errors' in value:
+                    zero_uncertainties = []
+                    for error in value['errors']:
+                        if 'symerror' in error:
+                            error_plus = error_minus = self.convert_to_float(
+                                error['symerror'],
+                                file_path=file_path,
+                                path=['dependent_variables', 'values', i, 'errors', 'symerror'],
+                                instance=data['dependent_variables']
+                            )
+                        elif 'asymerror' in error:
+                            error_plus = self.convert_to_float(
+                                error['asymerror']['plus'],
+                                file_path=file_path,
+                                path=['dependent_variables', 'values', i, 'errors', 'asymerror', 'plus'],
+                                instance=data['dependent_variables']
+                            )
+                            error_minus = self.convert_to_float(
+                                error['asymerror']['minus'],
+                                file_path=file_path,
+                                path=['dependent_variables', 'values', i, 'errors', 'asymerror', 'minus'],
+                                instance=data['dependent_variables']
+                            )
+                            if error_plus == '' and error_minus == '':
                                 error = ValidationError(
-                                    "Uncertainties should not all be zero",
-                                     path=['dependent_variables', 'values', i, 'errors'],
-                                     instance=data['dependent_variables']
+                                    "asymerror plus and minus cannot both be empty",
+                                    path=['dependent_variables', 'values', i, 'errors', 'asymerror'],
+                                    instance=data['dependent_variables']
```

Comment:
Can you please also invalidate `symerror: ''` (an empty string)?  That case is equivalent to an `asymerror` with `plus` and `minus` both set to an empty string.